packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2719,6 +2750,22 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
   /// possible that the node should be entirely removed.
   SemanticsAnnotator get semanticsAnnotator => null;
 
+  /// Assemble the [SemanticsNode] for this [RenderObject].
+  ///
+  /// If [isSemanticBoundary] is true, this method is called with the semantics
+  /// [node] created for this [RenderObject] and its semantics [children].
+  /// By default, the method will annotate [node] with the [semanticsAnnotator]
+  /// and add the [children] to it.
+  ///
+  /// Subclasses can override this method to add additional [SemanticNode]s
+  /// to the tree. If you do that, you also need to override [resetSemantics].
```

Comment:
elaborate here, "`If you do that, you also need to override [resetSemantics] to...`"